lib/amo-client.js

```diff
@@ -110,7 +110,12 @@ Client.prototype.waitForSignedAddon = function(statusUrl, opt) {
     function checkSignedStatus() {
       self.get({url: statusUrl}).then(function(result) {
         var data = result[1];
-        if (data.processed) {
+
+        var failedValidation = !data.valid;
+        // The add-on passes validation and all files have been created.
+        var passedValidation = data.valid && data.reviewed;
```

Comment:
This will poll for days if the add-on is listed. I guess you'd need to see that in the response.

I should add an `automated_signing` field. I think a reviewer can flag an unlisted add-on as needing manual review and it will never get auto-signed. In the future I guess `automated_signing` could flip from `true` to `false` if it didn't pass auto validation but the developer opted for manual review.